### PR TITLE
[#1045] Automated testing of the web socket

### LINKED ISSUE #1045: Automated test of the Websocket
It should be possible to test the WebSocket independently from the other parts with Cypress sending requests to the API and this [plugin](https://github.com/lensesio/cypress-websocket-testing).

### PR DESCRIPTION
**Pull request by ogbeche77**
Fixes #1045

### CODE REVIEW

File: integration/chat-plugin/BUILD [hunk 1]
```diff
@@ -8,6 +8,7 @@ ts_web_library(
     srcs = glob(["*.spec.ts"]),
     deps = [
         "//frontend/chat-plugin/handles:chat-plugin-handles",
+        "//frontend/ui/handles",
```
Comment (pascal-airy) [inline integration/chat-plugin/BUILD:11]: we don't need this dependency

File: integration/cypress.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
        "tagName": "Cypress Tag",
        "searchQuery": "Cypress Filter",
        "messageInbox": "Hello from Cypress Inbox!",
-       "messageChatplugin": "Hello from Cypress Chatplugin!",
+       "messageChatplugin": "Hello from Cypress ChatPlugin, Today is Beautiful!",
```
Comment (pascal-airy) [inline integration/cypress.json:12]: I like the positive mindset but maybe better to revert that change :)

File: integration/chat-plugin/websocket_test.spec.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+              text: 'Wonderful Day',
+            },
+          },
+        }).then(response => {
+          expect(response).property('status').to.equal(200);
+        });
+        cy.wait(3000);
+
+        cy.get(`[data-cy=${cyChatPluginListChat}]`).children().its('length').should('eq', 3);
+        cy.get(`[data-cy=${cyChatPluginListChat}]`).filter(':contains("Wonderful Day")');
```
Comment (pascal-airy) [inline integration/chat-plugin/websocket_test.spec.ts:81]: this is not an assertion. Assertions always have `should` in them. I think the assertion before is good enough so you can delete that whole line

File: integration/chat-plugin/websocket_test.spec.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    cy.visit('/channels');
+    cy.wait(500);
+    cy.url().should('include', '/channels');
+    cy.get(`[data-cy=${cyChannelsChatPluginAddButton}]`).click();
+    cy.get(`[data-cy=${cyChannelsChatPluginConnectButton}]`).click();
+    cy.get(`[data-cy=${cyChannelsChatPluginFormNameInput}]`).type(Cypress.env('chatPluginName'));
+    cy.get(`[data-cy=${cyChannelsChatPluginFormSubmitButton}]`).click();
+
+    cy.visit('http://localhost:8080/chatplugin/ui/example?channel_id=' + Cypress.env('channelId'));
```
Comment (pascal-airy) [inline integration/chat-plugin/websocket_test.spec.ts:30]: it is not good to have `localhost` in the test

File: frontend/chat-plugin/handles/index.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
 export const cyBubble = 'bubble';
 export const cyInputbarTextarea = 'inputbarTextarea';
 export const cyInputbarButton = 'inputbarButton';
+export const cyChatPluginListChat = 'chatPluginListChat';
```
Comment (pascal-airy) [inline frontend/chat-plugin/handles/index.ts:4]: naming is hard but important we should make clear that this is the message list. Maybe chatPluginMessageList ?

File: integration/ui/create_tag.spec.ts [hunk 6]
```diff
@@ -12,7 +12,6 @@ import {
   cyTagsTableRowDisplayDeleteModal,
   cyTagsTableRowDisplayDeleteModalInput,
   cyTagsTableRowDisplayDeleteModalButton,
-  cyChannelsChatPluginList,
```
Comment (pascal-airy) [inline integration/ui/create_tag.spec.ts:15]: good catch!

File: integration/chat-plugin/websocket_test.spec.ts [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    cy.get(`[data-cy=${cyInputbarButton}]`).click();
+
+    cy.wait(500);
+
+    cy.get(`[data-cy=${cyChatPluginListChat}]`).children().its('length').should('eq', 2);
+
+    cy.wait(2500);
+
+    cy.request('POST', 'http://airy.core/users.login', {
+      email: 'grace@example.com',
```
Comment (pascal-airy) [inline integration/chat-plugin/websocket_test.spec.ts:42]: you can also get the login information from the env vars

File: integration/chat-plugin/websocket_test.spec.ts [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+import {
+  cyChannelsChatPluginAddButton,
+  cyChannelsChatPluginConnectButton,
+  cyChannelsChatPluginFormNameInput,
+  cyChannelsChatPluginFormSubmitButton,
+} from 'handles';
+
+describe('Create UI Conversation', () => {
+  it('Creates a chat plugin conversation', () => {
```
Comment (pascal-airy) [inline integration/chat-plugin/websocket_test.spec.ts:11]: Please give the tests proper name and description

File: integration/chat-plugin/websocket_test.spec.ts [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+          method: 'POST',
+          url: 'http://airy.core/messages.send',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${loginToken}`,
+          },
+          body: {
+            conversation_id: `${conversationId}`,
+            message: {
+              text: 'Wonderful Day',
```
Comment (pascal-airy) [inline integration/chat-plugin/websocket_test.spec.ts:72]: lets also put this message in the env vars and make it more descriptive like "Websocket Test Message"

File: integration/chat-plugin/BUILD (hunk 1 above)
Comment (ogbeche77) [reply integration/chat-plugin/BUILD]: The build failed when I removed this dependency

Comment (pascal-airy) [reply integration/chat-plugin/BUILD:11]: right sorry!

**Review by pascal-airy — APPROVED** [review]
lgtm!
We will keep improving with https://github.com/airyhq/airy/issues/1386
but currently, there are more important things to test 🎉

**Pull request merged**